src/server/game/Server/WorldSession.h

```diff
@@ -348,8 +348,56 @@ class WorldSession
     void SendPacket(WorldPacket const* packet);
     void SendPetNameInvalid(uint32 error, std::string const& name, DeclinedName* declinedName);
     void SendPartyResult(PartyOperation operation, std::string const& member, PartyResult res, uint32 val = 0);
-    void SendAreaTriggerMessage(const char* Text, ...) ATTR_PRINTF(2, 3);
-    void SendAreaTriggerMessage(uint32 entry, ...);
+
+    template<typename... Args>
+    void SendAreaTriggerMessage(char const* Text, Args&&... args)
+    {
+        char szStr[1024];
```

Comment:
Is a fixed size buffer required? If not, this can be simplified:
```
    void SendAreaTriggerMessage(char const* Text, Args&&... args)
    {
        // Use Acore::StringFormat to generate the formatted string
        std::string formattedStr = Acore::StringFormat(Text, std::forward<Args>(args)...);
        uint32 length = formattedStr.size() + 1;
        ...

    }
```

Reply:
The old code had this fixed buffer.

```
va_list ap;
char szStr [1024];
szStr[0] = '\0';
```

I tried to be a close to the old code as possible. But if we can get rid of it, i will simplify it. Thanks for the feedback :)